Userland/Libraries/LibX86/Types/IdentView.h

```diff
@@ -0,0 +1,154 @@
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "IdentMap.h"
+#include <AK/Concepts.h>
+#include <AK/StdLibExtraDetails.h>
+#include <AK/Types.h>
+#include <string.h>
+
+#define MAKE_OPERATOR(op)                                                  \
+    template<Integral R, Integral U>                                       \
+    ALWAYS_INLINE IdentMap<R> operator op(const U& other) const            \
+    {                                                                      \
+        IdentMap<R> ret(m_size);                                           \
+        for (off_t i = 0; i < m_size; ++i) {                               \
+            ret[i] = m_values[i] op other;                                 \
+        }                                                                  \
+        return ret;                                                        \
+    }                                                                      \
+    template<Integral R, Integral U>                                       \
+    ALWAYS_INLINE IdentMap<R> operator op(const IdentMap<U>& other) const  \
+    {                                                                      \
+        VERIFY(m_size == other.size());                                    \
+        IdentMap<R> ret(m_size);                                           \
+        for (off_t i = 0; i < m_size; ++i) {                               \
+            ret[i] = m_values[i] op other[i];                              \
+        }                                                                  \
+        return ret;                                                        \
+    }                                                                      \
+    template<Integral R, Integral U>                                       \
+    ALWAYS_INLINE IdentMap<R> operator op(const IdentView<U>& other) const \
+    {                                                                      \
+        VERIFY(m_size == other.size());                                    \
+        IdentMap<R> ret(m_size);                                           \
+        for (off_t i = 0; i < m_size; ++i) {                               \
+            ret[i] = m_values[i] op other[i];                              \
+        }                                                                  \
+        return ret;                                                        \
+    }                                                                      \
```

Comment:
Both `IdentView.h` and `IdentMap.h` are very similar and `#include` each other.

These highlighted functions are identical in both files. Are these supposed to be `ALWAYS_INLINE IdentView<R> ...` rather than `ALWAYS_INLINE IdentMap<R> ...` ?

Reply:
In this  its a map, because it uses its own copy of data, instead of acting upon data provided to it

Reply:
And when doing operations we do not want to alter the data directly so its a Map there